Explorer/Assets/Scripts/ECS/StreamableLoading/Common/AssetPromise.cs

```diff
@@ -40,6 +40,15 @@ public struct AssetPromise<TAsset, TLoadingIntention> : IEquatable<AssetPromise<
                 Entity = world.Reference(world.Create(loadingIntention, partition, new StreamableLoadingState())),
             };
 
+        public static AssetPromise<TAsset, TLoadingIntention> CreateFinalized(World world, TLoadingIntention loadingIntention, IPartitionComponent partition,
+            StreamableLoadingResult<TAsset>? result) =>
+            new ()
+            {
+                LoadingIntention = loadingIntention,
+                Entity = world.Reference(world.Create(loadingIntention, partition, new StreamableLoadingState { Value = StreamableLoadingState.Status.Finished })),
```

Comment:
Creating an entity is redundant: just set it to `EntityRefence.Null` - this is exactly what's done when the promise is consumed